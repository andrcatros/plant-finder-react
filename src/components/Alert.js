import React from "react";
import PropTypes from "prop-types";

import StyledAlert from "../styles/styled-alert";

const Alert = ({ message, success }) => {
  if (!message) {
    return null;
  }

  return (
    <StyledAlert color={success ? "green" : "red"}>{message}!</StyledAlert>
  );
};

Alert.propTypes = {
  success: PropTypes.bool,
  message: PropTypes.string.isRequired,
};

Alert.defaultProps = {
  success: false,
};

export default Alert;
