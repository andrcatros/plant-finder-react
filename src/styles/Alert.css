.Alert {
  margin: auto;
  color: white;
  font-size: 16px;
  border-style: solid;
  border-radius: 10px;
  padding: 5px;
  font-weight: 550;
  text-align: center;
  width: 80%;
  margin-top: 50px;
}
