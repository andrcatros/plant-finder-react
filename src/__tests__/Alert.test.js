import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Alert from "../components/Alert";

describe("renders Alert component", () => {
  afterEach(cleanup);

  xit("renders error Alert component", () => {
    const { getByText, asFragment } = render(<Alert message="Error" />);
    expect(getByText(/Error/i).textContent).toBe("Error!");
    expect(asFragment()).toMatchSnapshot();
  });

  xit("renders success Alert component", () => {
    const { getByText, asFragment } = render(
      <Alert message="Success" success />
    );
    expect(getByText(/Success/i).textContent).toBe("Success!");
    expect(asFragment()).toMatchSnapshot();
  });
  it("does not render the Alert component when there is no message", () => {
    const { asFragment } = render(<Alert message="" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
