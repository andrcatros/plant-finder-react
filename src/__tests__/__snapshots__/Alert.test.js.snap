// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Alert component does not render the Alert component when there is no message 1`] = `<DocumentFragment />`;

exports[`renders Alert component renders error Alert component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iHdWFy"
    color="red"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`renders Alert component renders success Alert component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hHJyom"
    color="green"
  >
    Success!
  </div>
</DocumentFragment>
`;
